compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/XMLParser.java

```diff
@@ -477,7 +477,13 @@ private STNode parseXMLText() {
      * @return XML char-data token
      */
     private STNode parseCharData() {
-        return consume();
+        STToken token = consume();
+        if (token.kind != SyntaxKind.XML_TEXT_CONTENT) {
+            return STNodeFactory.createLiteralValueToken(SyntaxKind.XML_TEXT_CONTENT, token.text(),
+                    token.leadingMinutiae(), token.trailingMinutiae());
```

Comment:
When we create the new token out of the existing one, we have to propagate the diagnostics as well. (In case the the token already had error attached from the lexer).

```suggestion
            return STNodeFactory.createLiteralValueToken(SyntaxKind.XML_TEXT_CONTENT, token.text(),
                    token.leadingMinutiae(), token.trailingMinutiae(), token.diagnostics());
```

Reply:
Fixed